frontend/src/components/EditQuestion.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import TagsInput from 'react-tagsinput';
+import 'react-tagsinput/react-tagsinput.css';
+import { editQuestion, getQuestionDetails } from '../api/QuestionApi.js';
+import { EditWindow } from './ConfirmationWindow/ConfirmationWindows.js';
+import { showValidationErrorToast, showServerErrorToast, showSuccessToast, showFailureToast, showDuplicateQuestionErrorToast } from '../utils/toast.js';
+import TextEditor from './TextEditor/TextEditor.js';
+import '../css/Tags.css';
+
+const EditQuestion = () => {
+
+  const [newTitleValue, setTitleValue] = useState('');
+  const [newComplexityValue, setComplexityValue] = useState('');
+  const [newDescriptionValue, setDescriptionValue] = useState('');
+  const [newTags, setTags] = useState([]);
+  const [tagInput, setTagInput] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+  const [isEditWindowOpen, setEditWindowOpen] = useState(false);
+
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const question = await getQuestionDetails(id);
+        setTitleValue(question.title);
+        setComplexityValue(question.complexity);
+        setDescriptionValue(question.description);
+        setTags(question.tags);
+        setIsLoading(false);
+      } catch (error) {
+        navigate('../');
+        if (error.response.status === 400) {
+          showValidationErrorToast(error);
+        } else {
+          showServerErrorToast(error);
+        }
+      }
+    };
+
+    fetchData();
+  }, [id, navigate]);
+
+  const handleBackClick = () => {
+    setEditWindowOpen(true);
+  };
+
+  const handleConfirmQuit = () => {
+    setEditWindowOpen(false);
+    navigate(-1);
+  };
+
+  const handleEditWindowClose = () => {
+    setEditWindowOpen(false);
+  };
+
+  const handleSaveClick = async (e) => {
+    e.preventDefault();
+
+    try {
+      await editQuestion(id, newTitleValue, newComplexityValue, newDescriptionValue, newTags);
+      navigate(-1);
```

Comment:
(Mentioned in the other PR)
What does `-1` mean over here? Does it have a special meaning when provided to `navigate`?

Reply:
navigate(-1) basically means to return to the previous page you have accessed from.

E.g., I am currently at the path "http://localhost:3001/question/xxx"

If I were to click on the edit button, the path would bring me to "http://localhost:3001/edit/xxx"

Saving and using navigate(-1) would return me to "http://localhost:3001/question/xxx"

Thus we don't have to specify what is the path we want to return to for use cases such as back button.

Also, we do not want to use navigate('../') as this would navigate us straight to the parent route or path.